src/EndpointManager.sol

```diff
@@ -476,39 +561,94 @@ abstract contract EndpointManager is
 
         address transferRecipient = bytesToAddress(nativeTokenTransfer.to);
 
+        // Check inbound rate limits
+        bool isRateLimited = _isInboundAmountRateLimited(nativeTransferAmount, message.chainId);
+        if (isRateLimited) {
+            // queue up the transfer
+            uint64 queueSequence = _useInboundQueueSequence();
+
+            _getInboundQueueStorage()[queueSequence] = InboundQueuedTransfer({
+                amount: nativeTransferAmount,
+                recipient: transferRecipient,
+                txTimestamp: block.timestamp
+            });
```

Comment:
it may be good add a check here that the same message has not been added to the queue more than once. this function does have replay protection above, but this is a good defense in depth measure regardless

Reply:
This means we'd need to have additional replay protection for adding messages to queues, right?
Or perhaps we could simply replay-protect by ensuring that a number less than the `queueSequence` isn't being added to the queue? Since `queueSequence` monotonically increases

Reply:
Personally feel like the check is unnecessary because we're incrementing the `queueSequence` in advance right?

Reply:
That's true -- the check would be redundant.

I also made the outbound enqueue logic safer by bundling the `queueSequence` increment inside of the function -- so that's equivalent to making the check.

Reply:
ok that sounds good to me